src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -238,6 +238,11 @@ terms.
   return f
 end
 
+# TODO: is there a way to do this without manually dispatching for every non-conservative flux? We use this for the evaluation of boundary conditions.
+# if only one normal is specified, use that for `normal_direction_average`
+@inline flux_nonconservative_powell(u_ll, u_rr, normal_direction::AbstractVector, equations::IdealGlmMhdEquations2D) =
+  flux_nonconservative_powell(u_ll, u_rr, normal_direction, normal_direction, equations)
```

Comment:
This change was necessary to deal with `BoundaryConditionDirichlet`, which evaluates `surface_flux_function(u_inner, u_boundary, normal_direction, equations)` directly.

Reply:
I think so. We could dispatch on `have_nonconservative_terms(equations)` within `BoundaryConditionDirichlet` as well.

Reply:
Might be worth a try to see whether it can be cleaner

Reply:
I'll give that a shot

Reply:
I've incorporated the changes in https://github.com/trixi-framework/Trixi.jl/pull/1431/commits/6a184afb3fa4663e4fcd1e37a8780d7cdcd6121f by adding specializations `BoundaryConditionDirichlet` and `BoundaryConditionDoNothing` with a `using_nonconservative_flux::True` argument. It seems to work OK.

The only downside is that if you define a custom boundary condition, you now need to define two versions - one which takes in a symmetric flux and one which takes in `using_nonconservative_flux::True` and a non-symmetric flux. 

https://github.com/jlchan/Trixi.jl/blob/6a184afb3fa4663e4fcd1e37a8780d7cdcd6121f/examples/dgmulti_2d/elixir_mhd_reflective_BCs.jl#L68-L86

I find this kind of clunky.